sched/distributor/poolDistributor_test.go

```diff
@@ -1,230 +1,127 @@
-package distributor
+package distributor_test
 
 import (
-	"github.com/golang/mock/gomock"
-	cm "github.com/scootdev/scoot/sched/clustermembership"
-	"strings"
+	"github.com/scootdev/scoot/cloud/cluster"
+	"github.com/scootdev/scoot/cloud/cluster/memory"
+	"github.com/scootdev/scoot/sched/distributor"
 	"testing"
 )
 
-func TestPoolDistributor_NilCluster(t *testing.T) {
-	mockCtrl := gomock.NewController(t)
-	defer mockCtrl.Finish()
-
-	clusterMock := cm.NewMockCluster(mockCtrl)
-	clusterMock.EXPECT().Members().Return(nil)
-
-	dist := NewPoolDistributor(clusterMock)
-
-	if nil != dist {
-		t.Error("Expected dist to be nil when no members in Cluster")
-	}
-}
-
-func TestPoolDistributor_EmptyCluster(t *testing.T) {
-	mockCtrl := gomock.NewController(t)
-	defer mockCtrl.Finish()
-
-	nodes := make([]cm.Node, 0)
-
-	clusterMock := cm.NewMockCluster(mockCtrl)
-	clusterMock.EXPECT().Members().Return(nodes)
-
-	dist := NewPoolDistributor(clusterMock)
-
-	if nil != dist {
-		t.Error("Expected dist to be nil when no members in Cluster")
-	}
-}
-
 func TestPoolDistributor_OneNodeCluster(t *testing.T) {
-	mockCtrl := gomock.NewController(t)
-	defer mockCtrl.Finish()
-
-	nodes := cm.GenerateTestNodes(1)
-	clusterMock := cm.NewMockCluster(mockCtrl)
-	clusterMock.EXPECT().Members().Return(nodes)
-
-	dist := NewPoolDistributor(clusterMock)
-
-	if nil == dist {
-		t.Error("Expected dist to not be nil when 1 or more nodes in Cluster")
-	}
-
-	node1 := dist.ReserveNode()
-
-	select {
-	case <-dist.freeCh:
-		t.Error("Expected cluster to be full schedlued")
-	default:
-	}
-
-	dist.ReleaseNode(node1)
-	node2 := dist.ReserveNode()
-
-	if node1.Id() != node2.Id() {
-		t.Error("Expected nodes to be the same Id since there is only one in the cluster")
-	}
-
-	select {
-	case <-dist.freeCh:
-		t.Error("Expected cluster to be full schedlued")
-	default:
-	}
+	h := makeHelper(t, "node1")
```

Comment:
While this certainly makes the tests much clearer, I think the original mock tests could make use of a helper as well to greatly improve readability. That said, I don't think it would ever be quite as clean as this (which might be ok as long as it's close.)